pkg/controller/servicemesh/webhooks/mutation/controlplane_test.go

```diff
@@ -122,6 +131,13 @@ func TestCreate(t *testing.T) {
 
 				mutatedControlPlane := controlPlane.DeepCopy()
 				mutatedControlPlane.Spec.Profiles = []string{maistrav1.DefaultTemplate}
+				mutatedControlPlane.Spec.Gateways = &maistrav2.GatewaysConfig{
```

Comment:
instead of comparing the admissionrequests, why not compare the `mutatedControlplane` with the result of applying the patch? you can apply a patch using the github.com/evanphx/json-patch module (we already import it - in fact, gomodules.xyz/jsonpatch uses it for testing, too)